### PR TITLE
[MRG] ENH: func to extract list of instances of a certain entity from BIDS ds 

### LINKED ISSUE #250: get list of entities
Here's my Saturday night hack:

```py
# Authors: Mainak Jas <mainakjas@gmail.com>

import re
from pathlib import Path


def get_list_of_entity(bids_root, key='sub'):
    """Get list of a particular entity."""
    if key not in ('sub', 'ses', 'run', 'acq'):
        raise ValueError('Key must be one of. Got %s' % key)

    p = re.compile(r'%s-(.*?)_' % key)
    entities = list()
    for filename in Path(bids_root).rglob('*%s-*_*' % key):
        match = p.search(filename.stem)
        entity = match.group(1)
        if entity not in entities:
            entities.append(entity)
    return entities


bids_root = '/home/mainak/data/ds000248/'
print(get_list_of_entity(bids_root, 'sub'))
print(get_list_of_entity(bids_root, 'run'))
print(get_list_of_entity(bids_root, 'acq'))
```

you will get

```py
['01', 'emptyroom']
['01']
['flipangle05', 'flipangle30', 'epi']
```

can help us avoid `pybids` dependency

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #250 

I took the opportunity and refactored one test with the new pytest-kung-fu that I picked up from Marijn and Joan :)

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

File: mne_bids/tests/test_utils.py [hunk 1]
```diff
@@ -34,6 +34,40 @@
...
+def test_get_list_of_entity(return_bids_test_dir):
+    """Test getting a list of entities."""
+    bids_root = return_bids_test_dir
+    with pytest.raises(ValueError, match='Key must be one of'):
+        get_list_of_entity(bids_root, key='bogus')
+
+    assert get_list_of_entity(bids_root, 'sub') == [subject_id]
+    assert get_list_of_entity(bids_root, 'ses') == [session_id]
+    assert get_list_of_entity(bids_root, 'run') == [run]
+    assert get_list_of_entity(bids_root, 'acq') == [acq]
```
Comment (jasmainak) [inline mne_bids/tests/test_utils.py]: what happens if there is no `acq` key? or if there are more than one run? should we add a test for those cases?

File: mne_bids/utils.py [hunk 2]
```diff
@@ -28,6 +29,39 @@
 from mne_bids.tsv_handler import _to_tsv, _tsv_to_str
 
 
+def get_list_of_entity(bids_root, key='sub'):
```
Comment (jasmainak) [inline mne_bids/utils.py]: thinking about it, I think `key` should be a compulsory argument.

**Comment by jasmainak** [comment]
Appveyor is not happy ... is it related to this PR?

**Comment by sappelhoff** [comment]
No, we need to wait for  #814

**Comment by jasmainak** [comment]
okay let me know when it's ready

**Comment by jasmainak** [comment]
@sappelhoff in the meanwhile can you open a pull request to study template repo to use these instead of `pybids` ?

**Comment by sappelhoff** [comment]
@jasmainak this is ready to be merged. I pinned the CIs to a commit in BIDS-validator before the breakage (`a7e2fddeec489a8d5db81c561a5adc89363a306e`)

Once https://github.com/bids-standard/bids-validator/issues/814 is solved, we can go back to `master` of BIDS-validator

**Comment by sappelhoff** [comment]
> @sappelhoff in the meanwhile can you open a pull request to study template repo to use these instead of pybids ?

see: https://github.com/mne-tools/mne-study-template/pull/42

**Comment by jasmainak** [comment]
Thanks @sappelhoff !

**Pull request merged**